server/app/Icepeak/Main.hs

```diff
@@ -45,33 +47,35 @@ installHandlers core =
 
 main :: IO ()
 main = do
-  -- make sure output is flushed regularly
-  hSetBuffering stdout LineBuffering
+  crashLogger <- SentryLogging.getCrashLogger
+  SentryLogging.runWithCrashLogger "Control loop error" crashLogger $ do
+    -- make sure output is flushed regularly
+    hSetBuffering stdout LineBuffering
 
-  env <- getEnvironment
-  config <- execParser (configInfo env)
+    env <- getEnvironment
+    config <- execParser (configInfo env)
 
-  -- start logging as early as possible
-  logger <- Logger.newLogger (fromIntegral $ configQueueCapacity config)
-  loggerThread <- Async.async $ Logger.processLogRecords logger
+    -- start logging as early as possible
+    logger <- Logger.newLogger (fromIntegral $ configQueueCapacity config)
+    loggerThread <- Async.async $ Logger.processLogRecords logger
 
-  -- setup metrics if enabled
-  icepeakMetrics <- forM (configMetricsEndpoint config) $ const $ do
-    void $ Prometheus.register Prometheus.Metric.GHC.ghcMetrics
-    Metrics.createAndRegisterIcepeakMetrics
+    -- setup metrics if enabled
+    icepeakMetrics <- forM (configMetricsEndpoint config) $ const $ do
+      void $ Prometheus.register Prometheus.Metric.GHC.ghcMetrics
+      Metrics.createAndRegisterIcepeakMetrics
 
-  eitherCore <- Core.newCore config logger icepeakMetrics
-  either putStrLn runCore eitherCore
+    eitherCore <- Core.newCore config logger icepeakMetrics
+    either putStrLn (runCore crashLogger) eitherCore
```

Comment:
This `putStrLn` is probably an oversight and should also go via the logger. Could you fix it as well?